deps/src/main/java/com/microsoft/azure/sdk/iot/deps/transport/amqp/ErrorLoggingBaseHandler.java

```diff
@@ -8,6 +8,7 @@
 import lombok.extern.slf4j.Slf4j;
 import org.apache.qpid.proton.engine.BaseHandler;
 import org.apache.qpid.proton.engine.Event;
+import org.apache.qpid.proton.engine.Link;
 
 @Slf4j
 public class ErrorLoggingBaseHandler extends BaseHandler
```

Comment:
I think I'll just take this new logic I'm adding and make it into a new class that extends ErrorLoggingBaseHandler. I want the device client amqp layer to use the existing error logging base handler, but I want provisioning device client and iot service client to use this new logic since they both only ever have a single link in a single session in a single connection